### PR TITLE
Feat/reorg long term caching

### PR DESCRIPTION
**Pull request by tomchentw**
## Depends on

~~#10~~

## Description

* Can now enable long-term caching for assets folder under `/assrts/`
  - Can set expiration time to
* Link resources with `webpack` 
* Change folder structure
  - `dist` now contains all static files for deployment
  - `src` are all files that'll be processed by webpack

### CODE REVIEW

File: webpack.config.babel.js [hunk 1]
```diff
@@ -13,14 +13,14 @@ const DEFAULT_SITE = PRODUCTION_MODE ? 'wsstelnet://ws.ptt.cc/bbs' : 'wstelnet:/
...
+      './src/js/main.js',
+      './src/css/main.css',
+      './src/css/color.css'
     ]
   },
   output: {
-    path: path.join(__dirname, 'dist/'),
-    publicPath: '/dist/',
+    path: path.join(__dirname, 'dist/assets/'),
+    publicPath: '/assets/',
```
Comment (robertabcd) [inline webpack.config.babel.js]: Can you use relative path ('assets/') so the site can be put into non-root directory?
I was about to push a commit for this. Maybe you can do this along with this PR. Thanks!

Comment (tomchentw) [reply webpack.config.babel.js]: Good idea! I've done it in the last commit and rebased.
Testes development env and it's working fine. Could you test prod env on local?

Comment (robertabcd) [reply webpack.config.babel.js]: It looks like I always build prod and use ?site= to override the connecting site. It looks fine to me.

Comment (tomchentw) [reply webpack.config.babel.js]: Yeah b/c I didn't setup a nginx proxy on local (just use webpack-dev-server's proxy). Thanks for the help!

**Pull request merged**